File: src/main/java/com/elemonated/networker/persistence/data/Employee.java
package com.elemonated.networker.persistence.data;

import lombok.Data;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;

@Data
@Entity
public class Employee {

    @Id
    @GeneratedValue
    private Long id;

    private String firstName;

    private String lastName;

    @Column(unique = true, nullable = false)
    private String email;

    private String phone;

}

File: net-worker-ui/src/app/model/employee.ts
export type EmployeeId = number;

export interface Employee {
  id?: EmployeeId;

  firstName?: string;

  lastName?: string;

  email?: string;

  phone?: string;
}

export function getReadableEmployeeName(employee: Employee): string {
  const firstNamePart = employee.firstName ? employee.firstName + " " : "";
  const lastNamePart = employee.lastName ? employee.lastName : "";
  return firstNamePart + lastNamePart;
}
